api/cluster.go

```diff
@@ -563,6 +563,12 @@ func queryPeers(ctx context.Context, peerGroups map[int32][]cluster.Node, name s
 		}
 
 		for group, peers := range peerGroups {
+			if len(peers) == 0 {
+				errorChan <- fmt.Errorf("peer group %d has no peers", group)
```

Comment:
errorChan is a buffered chan of size 1. If we send to it here and continue, we have the chance of blocking on the channel [here](https://github.com/grafana/metrictank/pull/1747/files#diff-bc8a656be21edce0cf2a74adf23d7aeaR614). Since the common approach taken by callers is to consume all responses, then consume the errorChan, that could be a deadlock.

Also, if this is supposed to be an ignorable error (since we continue processing) maybe it should just be logged? There is no good way for a caller to know to ignore this error vs propagating it.

Reply:
right, good catch! i'll just log it